services/matrix/matrix.tester.js

```diff
@@ -0,0 +1,232 @@
+'use strict'
+
+const Joi = require('joi')
+const ServiceTester = require('../service-tester')
+
+const t = new ServiceTester({ id: 'matrix', title: 'Matrix' })
+module.exports = t
+
+t.create('get room state as guest')
+  .get('/ROOM/DUMMY.dumb.json?style=_shields_test')
+  .intercept(nock =>
+    nock('https://DUMMY.dumb/')
+      .post('/_matrix/client/r0/register?kind=guest')
+      .reply(
+        200,
+        JSON.stringify({
+          access_token: 'TOKEN',
+        })
+      )
+      .get('/_matrix/client/r0/rooms/ROOM:DUMMY.dumb/state?access_token=TOKEN')
+      .reply(
+        200,
+        JSON.stringify([
+          {
+            // valid user 1
+            type: 'm.room.member',
+            sender: '@user1:DUMMY.dumb',
+            state_key: '@user1:DUMMY.dumb',
+            content: {
+              membership: 'join',
+            },
+          },
+          {
+            // valid user 2
+            type: 'm.room.member',
+            sender: '@user2:DUMMY.dumb',
+            state_key: '@user2:DUMMY.dumb',
+            content: {
+              membership: 'join',
+            },
+          },
+          {
+            // should exclude banned/invited/left members
+            type: 'm.room.member',
+            sender: '@user3:DUMMY.dumb',
+            state_key: '@user3:DUMMY.dumb',
+            content: {
+              membership: 'leave',
+            },
+          },
+          {
+            // exclude events like the room name
+            type: 'm.room.name',
+            sender: '@user4:DUMMY.dumb',
+            state_key: '@user4:DUMMY.dumb',
+            content: {
+              membership: 'fake room',
+            },
+          },
+        ])
+      )
+  )
+  .expectJSON({
+    name: 'chat',
+    value: '2 users',
+    colorB: '#4c1',
```

Comment:
Instead of using hex codes for the colour tests, could you compare them to constants instead.

If you import

```js
const { colorScheme } = require('../test-helpers')
```

this can be written as `colorB: colorScheme.brightgreen` which is easier to understand than `colorB: '#4c1'` ( + so on for the other colour tests)